src/mlpack/tests/neat_test_tools.hpp

```diff
@@ -0,0 +1,251 @@
+/**
+ * @file neat_test_tools.hpp
+ * @author Rahul Ganesh Prabhu
+ *
+ * This file includes some useful classes for neat tests.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_TESTS_NEAT_TEST_TOOLS_HPP
+#define MLPACK_TESTS_NEAT_TEST_TOOLS_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/reinforcement_learning/environment/continuous_multiple_pole_cart.hpp>
+#include <mlpack/methods/reinforcement_learning/environment/multiple_pole_cart.hpp>
+#include <mlpack/methods/ann/activation_functions/hard_sigmoid_function.hpp>
+#include <mlpack/methods/neat/neat.hpp>
+
+using namespace mlpack;
+using namespace mlpack::neat;
+using namespace mlpack::ann;
+using namespace mlpack::rl;
+
+// A class that defines the XOR task.
+class XORTask
+{
+ public:
+  double Evaluate(Genome<HardSigmoidFunction>& genome)
+  {
+    arma::vec input1 = {0, 0};
+    arma::vec input2 = {0, 1};
+    arma::vec input3 = {1, 0};
+    arma::vec input4 = {1, 1};
+    double error = 0;
+    error += findError(input1, genome); 
+    error += findError(input2, genome); 
+    error += findError(input3, genome); 
+    error += findError(input4, genome);
+    return 4 - error;
+  }
+
+  double findError(arma::vec& input, Genome<HardSigmoidFunction>& genome)
+  {
+    arma::vec output = genome.Evaluate(input);
```

Comment:
To match with the existing codebase, I think it makes sense to allow the user to pass an ``arma::mat``and internally iterate over the columns, instead of doing this outside the method. In this case we could pass the complete input data and use ``arma::accu(arma::pow(...))`` to get the error.